src/main/java/picard/arrays/GtcToVcf.java

```diff
@@ -0,0 +1,688 @@
+            builder.attribute(InfiniumVcfFields.DEV_X_BB, formatFloatForVcf(bbVals.devX));
+            builder.attribute(InfiniumVcfFields.MEAN_X_AA, formatFloatForVcf(aaVals.meanX));
+            builder.attribute(InfiniumVcfFields.MEAN_X_AB, formatFloatForVcf(abVals.meanX));
+            builder.attribute(InfiniumVcfFields.MEAN_X_BB, formatFloatForVcf(bbVals.meanX));
+            builder.attribute(InfiniumVcfFields.DEV_Y_AA, formatFloatForVcf(aaVals.devY));
+            builder.attribute(InfiniumVcfFields.DEV_Y_AB, formatFloatForVcf(abVals.devY));
+            builder.attribute(InfiniumVcfFields.DEV_Y_BB, formatFloatForVcf(bbVals.devY));
+            builder.attribute(InfiniumVcfFields.MEAN_Y_AA, formatFloatForVcf(aaVals.meanY));
+            builder.attribute(InfiniumVcfFields.MEAN_Y_AB, formatFloatForVcf(abVals.meanY));
+            builder.attribute(InfiniumVcfFields.MEAN_Y_BB, formatFloatForVcf(bbVals.meanY));
+            if (zCallThresholds.containsKey(egtFile.rsNames[egtIndex])) {
+                String[] zThresh = zCallThresholds.get(egtFile.rsNames[egtIndex]);
+                builder.attribute(InfiniumVcfFields.ZTHRESH_X, zThresh[0]);
+                builder.attribute(InfiniumVcfFields.ZTHRESH_Y, zThresh[1]);
+            }
+            final String rsid = record.getRsId();
+            if (StringUtils.isNotEmpty(rsid)) {
+                builder.attribute(InfiniumVcfFields.RS_ID, rsid);
+            }
+            if (egtFile.totalScore[egtIndex] == 0.0) {
+                builder.filter(InfiniumVcfFields.ZEROED_OUT_ASSAY);
+                if (genotype.isCalled()) {
+                    throw new PicardException("Found a call on a zeroed out SNP!!");
+                }
+            }
+            if (record.isDupe()) {
+                builder.filter(InfiniumVcfFields.DUPE);
+            }
+
+            numVariantsWritten++;
+            contexts.add(builder.make());
+
+
+            gtcIndex++;
+        }
+
+        log.info(numVariantsWritten + " Variants were written to file");
+        log.info(gtcFile.getNumberOfSnps() + " SNPs in the GTC file");
+        log.info(manifest.getNumAssays() + " Variants on the " + manifest.getDescriptorFileName() + " genotyping array manifest file");
+    }
+
+    //Uses Manhattan distance conversion
+    private EuclideanValues polarToEuclidean(float r, float rDeviation, float theta, float thetaDeviation) {
+        //calculate variance (deviation^2)
+        final double thetaVariance = Math.pow(thetaDeviation, 2.0);
+        final double rVariance = Math.pow(rDeviation, 2.0);
+
+        final double halfPi = Math.PI / 2;
+        final double normalizedTheta = halfPi * theta;
+        final double rOverX = (1 + Math.tan(normalizedTheta));
+
+        //calculate X and Y variances from R and Theta variances
+        final double thetaVarianceFactorX = -1 * (halfPi * r) * Math.pow((1 + Math.tan(normalizedTheta)), -2) * (1 / Math.pow(Math.cos(normalizedTheta), 2));
```

Comment:
pow is an expensive function, so you can save on one of them here by converting to 
```suggestion
        final double thetaVarianceFactorX = -1 * (halfPi * r) * Math.pow((1 + Math.tan(normalizedTheta))  *Math.cos(normalizedTheta), -2));
```

Reply:
Done.